### ISSUE TITLE
Building Machines That Learn and Think Like People

### ISSUE THREAD

**Issue by omoindrot** [omoindrot/papers#2]
http://arxiv.org/pdf/1604.00289v1.pdf

> The central goal of this paper is to propose a set of core ingredients for building more human-like learning and thinking machines.

The authors of this paper focus only on human inspired machines, and try to infer the principles of human intelligence. Especially, most AI algorithms today use **pattern recognition** to perform tasks. The authors of the paper argue that human intelligence rely essentially on **model-building**

There are three core ingredients:  
- **start-up software**
  - cognitive capabilities present early in development
  - intuitive physics
  - intuitive psychology
- **learning**
  - learning is a form of model building, of understanding the causality of the world
  - compositionality
  - learning-to-learn
- **put these models into action**
  - humans react to new situations very quickly
  - model based methods are slow
  - humans combine model-based learning with model-free reinforcement learning, which provides fast responses to common situations
  - both mechanisms are coordinated by a _metacognitive process_
## 2. Cognitive and neural inspiration in Artificial Intelligence

Today the majority of algorithms in deep learning use a generic neural network, going from a blank state to accurate object recognition for instance. However, these algorithms may be enhanced to be more human like, as they only work with an immense amount of data. Human, on the other side can train themselves to recognize objects with very few training on data.
## 3. Challenges for building more human-like machines

The human brain is not composed of these generic neural networks, and cognitive science stresses out:

> the importance of early inductive biases, including core concepts such as number, space, agency and objects, as well as powerful learning algorithms that rely on prior knowledge to extract knowledge from small amounts of training data

The paper presents two example challenges to help illustrate the importance of core cognitive ingredients introduced in section 4.
### 3.1 The Characters Challenge

Computers achieve human-level accuracy on MNIST handwritten digits dataset, or even on the much harder object recognition ImageNet dataset. However, the paper argues that the deep networks trained on these data only use a complex pattern recognition, whereas humans learn a **concept** of the character, even with one example of it. They can then easily achieve a variety of tasks:
- Detect this character among other characters with only one training example
- Generate new examples from this one training example
- Parse a character into its most important parts
- Generate new characters using a small set of related characters

Progress on these 4 tasks has been made using **probabilistic program induction** (cf. [Human-level concept learning through probabilistic program induction](http://web.mit.edu/cocosci/Papers/Science-2015-Lake-1332-8.pdf))
### 3.2 The Frostbite Challenge

Frosbite is a classic Atari game that involves long term planning. Deep Q Learning performed by DeepMind had trouble on this game to achieve human-level score.  
The algorithm also uses a lot of gaming time (almost 1000 hours) to achieve its best score (around 80%), 
whereas the human only has 2 hours of gaming time.

![Frosbite score](https://cloud.githubusercontent.com/assets/13662086/14383422/4960de8e-fd95-11e5-9cf8-d6de9f2ceaf5.png)

Moreover, a human could easily achieve other goals in the game (e.g. not dying...) without retraining, whereas the DQN has to retrain from scratch with another goal function. These other goals might include:
- get the lowest possible score
- get close to a score
- die as quickly as possible
- get as many fish as possible

The model of the game humans learn is then much more **flexible**, and can generalize to other tasks instantaneously.
